tests/test_noninteractive_mode.py

```diff
@@ -0,0 +1,119 @@
+"""
+Non-interactive tests.
+"""
+import subprocess
+import os
+import pytest
+from mssqltestutils import (
+    create_mssql_cli,
+    create_test_db,
+    clean_up_test_db,
+)
+
+_FILE_ROOT = '%s/test_query_results/' % os.path.dirname(os.path.abspath(__file__))
+
+class TestNonInteractiveResults:
+    """
+    Tests non-interactive features.
+    """
+
+    testdata = [
+        ("SELECT * FROM STRING_SPLIT(REPLICATE(CAST('X,' AS VARCHAR(MAX)), 1024), ',')",
+         '%sbig.txt' % _FILE_ROOT),
+        ("SELECT 1", '%ssmall.txt' % _FILE_ROOT),
+        ("SELECT 1; SELECT 2;", '%smultiple.txt' % _FILE_ROOT),
+        ("SELECT %s" % ('x' * 250), '%scol_too_wide.txt' % _FILE_ROOT),
+        ("SELECT REPLICATE(CAST('X,' AS VARCHAR(MAX)), 1024)", '%scol_wide.txt' % _FILE_ROOT)
+    ]
+
+    @pytest.mark.parametrize("query_str, file_baseline", testdata)
+    def test_query(self, query_str, file_baseline):
+        output_query = self.execute_query_via_subprocess(query_str)
+        output_baseline = self.get_file_contents(file_baseline)
+        assert output_query == output_baseline
+
+    @staticmethod
+    def test_noninteractive_run():
+        """ Test that calling run throws an exception only when interactive_mode is false """
+        mssqlcli = create_mssql_cli(interactive_mode=False)
+        try:
+            mssqlcli.run()
+            assert False
+        except ValueError:
+            assert True
+        finally:
+            mssqlcli.shutdown()
+
+    @staticmethod
+    def execute_query_via_subprocess(query_str):
+        """ Helper method for running a query with -Q. """
+        p = subprocess.Popen("./mssql-cli -Q \"%s\"" % query_str, shell=True, stdin=subprocess.PIPE,
+                             stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        output = p.communicate()[0].decode("utf-8")
+        return output
+
+    @staticmethod
+    def get_file_contents(file_path):
+        """ Get expected result from file. """
+        try:
+            with open(file_path, 'r') as f:
+                return f.read()
+        except OSError as e:
+            raise e
+
+
+class TestNonInteractiveDbModification:
+    """
+    Test DB modification in non-interactive mode.
+    """
+
+    @staticmethod
+    @pytest.fixture(scope="class")
+    def test_db():
```

Comment:
Let's just remove the TestNonInteractiveDbModification test - they repeat the TestNonInteractiveShutdown code path.